lib/diagnostics.cpp

```diff
@@ -505,9 +509,9 @@ int diagnostics_init(
             // How large does our buffer need to be?
             dwSize = sizeof(boinc_install_dir);
 
-            lReturnValue = RegQueryValueEx(
+            lReturnValue = RegQueryValueExA(
```

Comment:
Why changing from Unicode function to ANSI?

Reply:
API correctness
`boinc_install_dir` is a `char` string
`RegQueryValueEx` is a generic-text function
The two should never be mixed

Reply:
`RegQueryValueEx` is a macro (since `RegQueryValueExW` was added), and where it depends on the project settings what it will be resolved to, `RegQueryValueExA` or `RegQueryValueExW`.
Currently it resolves to `RegQueryValueExA`.
I suggest to keep it as is because in case we decide to make it wide char, it would be much easier to change to `boinc_install_dir` value on build error, rather than looking for all functions with `A` suffix at the end.

Reply:
Changing to widechar is not going to be easy whichever way it’s done. In BOINC it could only reasonably be implemented in the context of ‘UTF-⁠8 everywhere’, in which case simply changing ANSI to Unicode and seeing what breaks will not even come close to getting it right. And the risk of that approach is that people will ‘fix’ the breakage by throwing in casts to make the compiler warnings go away while accusing Microsoft of not knowing what they’re doing… (Like [here](https://github.com/BOINC/boinc/blob/master/apps/concat.cpp#L95-L98 "apps/concat.cpp"), for instance.)

In theory, everything should _compile_ cleanly in both ANSI and Unicode. If it doesn’t, the code is using the API wrong. (Whether the application _works_ in both modes is an entirely different question…)

There are dozens of other examples of this inconsistency throughout BOINC; this instance only showed up as overlap between two concurrent patchsets which I haven’t kept in separate branches. The bigger question would be: are you interested in that kind of ‘code quality’ patch, or is it beyond the kind of thing that is worth spending time on?

Reply:
In this particular case I prefer to keep it as-is, and do not change macro to the function with `A` suffix.
Thank you for understanding.

Answering you bigger question, it depends: I can't answer for all cases, each should be reviewed separately.